File: 2232-adding-spaces-to-a-string/adding-spaces-to-a-string.java
class Solution {
    public String addSpaces(String s, int[] spaces) {
        StringBuilder ans = new StringBuilder();
        int i = 0;
        for (int j = 0; j < s.length(); j++) {
            if (i < spaces.length && spaces[i] == j) {
                ans.append(" ");
                i++;
            }
            ans.append(s.charAt(j));
        }
        return ans.toString();
    }
}

File: 2232-adding-spaces-to-a-string/adding-spaces-to-a-string.cpp
class Solution {
public:
    string addSpaces(string s, vector<int>& spaces) {
        int i = 0;
        string ans = "";
        for(int j=0;j<s.size();j++){
            if(i<spaces.size() && spaces[i] == j){
                ans+=" ";
                i++;
            }
            ans += s[j];
        }
        return ans;
    }
};